### ISSUE TITLE
Tag a new release version

### ISSUE THREAD

**Issue by szlin** [rigred/sandsifter#3]
Could you add changelog and make a new release with specific version (perhaps 1.0) ?

It would be appreciated if you could do this.

**Comment by szlin** [comment]
@rigred I suggest to change the path in below, do you have any thoughts about this?
```
INJECTOR = "./injector"
arch = ""

OUTPUT = "./data/"
```
--
INJECTOR = "/usr/bin/injector"
arch = ""

OUTPUT = "/tmp/sandsift/data/"

**Comment by rigred** [comment]
I will make a few changes to keep it as distro agnostic as possible. Logging to tmp is certainly possible, but I prefer a more non-volatile storage location since running sandsifter is something that can potentially cause a system lockup (It's implied in the nature of the program) so logging to a non-volatile location would be preferred as log data that could indicate where it failed is otherwise lost.

I'm not a debian user, but I suspect I am correct in my assessment that /tmp is treated as tmpfs?

My alternative suggestion would be to write to a user home directory or ask for explicit input on where to write log files, what do you think? 

Do you have an alternative suggestion?

**Comment by rigred** [comment]
I had a think about it and decided it would be best for sandsifter to require the user explicitly provide the output path. Since sandsifter needs to run as root / sudo it's not wise to make assumptions about where to write the data.

**Comment by rigred** [comment]
https://github.com/rigred/sandsifter/commit/c1bd38824c95493d3ef4daaf142121c37673cb3f

Added explicit output argument, defaults to `/tmp/sandsifter/data` if no `--out` is given.

**Comment by szlin** [comment]
@rigred The /tmp in Debian is *not* using tmpfs. However, it would be wiped out after booting up in default.
The reason why I suggest to put under /tmp is as you said the sandsifter needs permission of superuser, and thus I want to find a safer place for it.

**Comment by szlin** [comment]
@rigred Fair enough :)  https://github.com/rigred/sandsifter/commit/c1bd38824c95493d3ef4daaf142121c37673cb3f

**Comment by rigred** [comment]
Pushed released 1.01 with requested fixes. :)

**Comment by rigred** [comment]
Code now checks in `/usr/bin/injector` or `./injector` for sandsifter injector.

**Comment by rigred** [comment]
I will close this issue now, but feel free to open any enhancement requests and ideas you have.

**Comment by szlin** [comment]
@rigred Thanks for your prompt reply! I'll repackage new version tomorrow since it's midnight here :)

**Comment by szlin** [comment]
@rigred Do you plan to use setuptools / egg_info to build and distribute sandsifter?

**Comment by rigred** [comment]
I might add that, just need to try out first since I don't usually use that.

**Comment by szlin** [comment]
@rigred I have done the beta package in below.

IMHO, the name of binary "injector" and "summarize" are too common for causing naming conflict to others. Could you please consider to add some prefixes such as "sifter-injector" or "sifter-sum" to avoid the conflict?

/usr/bin/injector
/usr/bin/sifter
/usr/bin/summarize
/usr/lib/python2.7/dist-packages/sandsift/disas/capstone_32.py
/usr/lib/python2.7/dist-packages/sandsift/disas/capstone_64.py
/usr/lib/python2.7/dist-packages/sandsift/disas/disas_32.sh
/usr/lib/python2.7/dist-packages/sandsift/disas/disas_64.sh
/usr/lib/python2.7/dist-packages/sandsift/gui/__init__.py
/usr/lib/python2.7/dist-packages/sandsift/gui/gui.py
/usr/lib/python2.7/dist-packages/sandsift/pyutil/__init__.py
/usr/lib/python2.7/dist-packages/sandsift/pyutil/colors.py
/usr/lib/python2.7/dist-packages/sandsift/pyutil/progress.py
/usr/share/doc/sandsifter/README.md.gz

**Comment by rigred** [comment]
Yeah I considered doing that, It shall be done :D

**Comment by szlin** [comment]
@rigred Awesome, please ping me when you have done that.

**Comment by rigred** [comment]
@szlin small version increment to rename the files etc
https://github.com/rigred/sandsifter/releases/tag/v1.02

**Comment by rigred** [comment]
This is generally fine since there is little reason for running sandsifter as an unprivileged user.

**Comment by rhertzog** [comment]
FWIW, using /tmp for output data is precisely a bad idea when you are running root. The reasons we are using tools like "mktemp" to create temporary files is to avoid the possibility of race attacks where the attacker puts a symlink or a hardlink in place before the tool run by root does its job and overwrite some target file... so a fixed path in /tmp is a no-go.

Please consider using user files in $HOME... maybe in $HOME/.local/sandsifter/ (persistent data) or $HOME/.cache/sandsifter/ (data that can be cleared without adverse consequences).

**Comment by rigred** [comment]
I aim to make a number of changes made to sandsifter in the coming months. 

1. change the log file output to give the logs proper filenames. 
2. Add default (sane scanning) parameters.
3. convert the code to python 3 and perform script housekeeping.
4. Change the summarizer to:
     * Output a properly sorted and compressed instruction table.
     * Speed up disassembly.
     * Reduce memory load.
     * Improve summary usefulness with an outside api.
5. Attempt to speed up scans with chunked multi-threaded tunneling.
6. Actually find bugs

**Comment by rigred** [comment]
Just keep in mind that I'm quite busy otherwise and can only get around to this when free time allows.

**Comment by szlin** [comment]
@rigred One last thing to be confirmed:
Build-depends: libcapstone-dev, libcapstone3
Execution-depends: python-capstone, libcapstone3

Please correct me if I am wrong, thanks.

**Comment by rigred** [comment]
Correct.

Optionally there are also
`ndisasm` and `objdump` for the sifter/summarize.

**Comment by shirishag75** [comment]
Any idea when move to python3 will be done ?

**Comment by rigred** [comment]
@shirishag75 Branch https://github.com/rigred/sandsifter/tree/python3 ports this project to python3
It probably still has bugs so please give it a test.

**Comment by szlin** [comment]
@rigred I would like to inform you that package of sandsifter was accepted into Debian archive [1].

[1] https://packages.debian.org/sid/sandsifter
